Premier passe vers UP V.5

audreyazura opened the pull request:
Armes :
Suppression des armes supprimées après le passage à la V4.
Modifications des catégories.
Modifications des caracs des armes, à l'exclusion des valeurs de dégât.

Armures :
Passage sur les intervales de points d'armure, les réduction de dégâts, les bonus au ND, les noms et les valeurs en points d'armure de chacune des parties.

Commit:
caracs_armes.json :
Mise à jour vers le UP4.5
Arme lourde de càc : faire deux entrée pour avec/sans harnais
Suppression de la catégorie arme à chaîne
Mousqueton -> Mousquet
pistolet à rouet -> pistolet ancien

Commit:
caracs_armes.json :
Mise à jour vers le UP V.5, sauf dégât

Commit:
caracs_armes.json :
Type des armes lourdes corrigé

caracs-armures.json :
Rangs, reduction de dégât, bonus au ND et valeurs en points des armes mis à jours vers la V.5

Commit:
caracs_armes.json :
Type des armes lourdes corrigé

Commit:
Mise en place des tests

Commit:
Tous les tests passés

ykonoclast commented:
Ok super, les tests passent (sous réserve qu'ils n'aient pas été modifiés uniquement pour passer [i.e. en occultant des erreurs légitimes]) et la branche n'a pas de conflit.

Néanmoins je ne peux pas m'en occuper tout de suite et je ne veux pas merger une pull-request sur un coup de tête, donc j'y reviens dès que possible et je regarde les modifications dans le détail avant de l'amalgamer au tronc.

ykonoclast commented:
Bon j'ai vérifié et je ne peux pas la prendre en l'état 

1) tu inclues dans ton commit "Mise en place des tests" pas mal de gros fichiers de log (tous préfixés par "hs_err_pid"). Ils n'ont rien à faire dans la base de code, il faut que tu les en exclues.

2) Dans les tests tu te contentes de changer les résultats pour "que ça passe" mais parfois en faisant cela.... ben tu vides le test de son sens. Dans le test de perso par exemple on ajoute une pièce d'armure et on reteste le ND pour être touché pour vérifier qu'il a changé.... ben toi du coup tu testes qu'il est encore le même puisque l'échelle des armures a changé. Quand c'est comme ça il faut plutôt rajouter un test : convertir le résultat (oui ok ça ne change pas) mais derrière re-rajouter une pièce d'armure pour bien tester que la fonctionalité d'augmentation du ND par l'armure marche bien.
En l'état un perso du UP!System pourrait très bien ne recevoir aucun bonus d'armure : les tests le laisseraient passer. 
Tu ne peux pas juste changer les tests pour "les faire passer", il faut comprendre ce qu'ils testent et les enrichir ou a minima garder la fonctionalité testée, sinon c'est la porte ouverte aux régressions.

Du coup je suis désolé parce que tu as bien et beaucoup travaillé à mon bénéfice mais malheureusement si je dois passer plus de temps à tout relire et revérifier tous les tests que je n'en prendrais à les faire moi-même.... ben ce n'est pas trop porteur de valeur pour moi et je préfère à la limite que tu te limites aux armes.

audreyazura commented:
Ok, je suis désolé, je ne me suis pas rendu qu'à certains moments je faisais. Je suis allé un peu trop vite par moment ><. Je me mettrais sur la réécriture des tests demain, du coup.

Commit:
carac_armes :
Corrections de quelques fausses catégories

Test :
Remise en place des tests aux limites, ajout de quelques autres.

audreyazura committed:
Delete hs_err_pid10261.log

audreyazura committed:
Delete hs_err_pid10278.log

audreyazura committed:
Delete hs_err_pid10296.log

audreyazura committed:
Delete hs_err_pid10587.log

audreyazura committed:
Delete hs_err_pid6536.log

audreyazura committed:
Delete hs_err_pid6553.log

audreyazura committed:
Delete hs_err_pid6570.log

audreyazura committed:
Delete hs_err_pid7021.log

audreyazura committed:
Delete hs_err_pid7039.log

audreyazura committed:
Delete hs_err_pid7057.log

audreyazura committed:
Delete hs_err_pid7283.log

audreyazura committed:
Delete hs_err_pid7300.log

audreyazura committed:
Delete hs_err_pid7319.log

audreyazura committed:
Delete hs_err_pid7548.log

audreyazura committed:
Delete hs_err_pid7566.log

audreyazura committed:
Delete hs_err_pid7585.log

audreyazura committed:
Delete hs_err_pid7814.log

audreyazura committed:
Delete hs_err_pid9048.log

audreyazura committed:
Delete hs_err_pid9584.log

audreyazura committed:
Delete hs_err_pid9601.log

audreyazura committed:
Delete hs_err_pid9618.log

audreyazura committed:
Delete hs_err_pid9954.log

audreyazura committed:
Delete hs_err_pid9971.log

audreyazura committed:
Delete hs_err_pid9988.log

ykonoclast commented:
Hmmmmm,

 -	persoRM1.getInventaire().addPieceArmure(new PieceArmure(0, 0, 0, false), Inventaire.ZoneEmplacement.TETE);
 +	persoRM1.getInventaire().addPieceArmure(new PieceArmure(7, 0, 0, false), Inventaire.ZoneEmplacement.CORPS);
  	Assert.assertEquals(15, persoRM1.getNDPassif(0, 1, false));
 -	Assert.assertEquals(15, persoRM1.getNDPassif(0, 1, true));
 +	Assert.assertEquals(5, persoRM1.getNDPassif(0, 1, true));

puis

 -	persoRM1.getInventaire().removePieceArmure(Inventaire.ZoneEmplacement.TETE);
 +	persoRM1.getInventaire().removePieceArmure(Inventaire.ZoneEmplacement.CORPS);

Dans l'absolu tu as compris la critique précédente et tu continues de bien tester que le ND passif augmente avec une armure. Malheureusement, comme je te le disais, avec la nouvelle échelle tu as aussi introduit un nouveau cas limite : celui de la pièce d'armure qui n'a aucun effet (moins de deux points d'armure car maintenant ça commence à deux). 

Il ne fallait donc pas supprimer carrément ce pauvre casque mais le laisser faire son truc avec ta modification (on vérifie qu'effectivement tout seul il ne fait rien) et après tu peux rajouter ta cuirasse pour vérifier qu'alors le ND augmente bien.

Je suis conscient de paraître tatillon mais il faut comprendre que je ne peux pas me rappeler de tout ce que j'ai fait des mois auparavant, il m'a fallu un quart d'heure pour comprendre comment marchait ce système aujourd'hui (alors que je l'ai écrit!) donc, les tests sont la SEULE assurance que lorsque je change un truc l'ensemble continue de marcher. Surtout pour des trucs un peu silencieux comme le calcul des effets d'armure ou des dégâts. Il faut donc qu'ils soient absolument nickel, on ne peut pas les rendre inviolables mais il faut être certain qu'ils attraperont le plus d'erreurs possibles.

Il faudrait donc que ce test précis soit enrichi et pas juste modifié.

Commit:
PersoNonStatTest :
Remise en place du casque au perso RM1 pour tester les limites.

Commit:
Merge origin-fork/master

audreyazura commented:
Pas de soucis. J'ai remis les tests avec le casque, du coup, et bien pris soin de l'enlever ensuite. Tout devrait être bon, du coup.

ykonoclast commented:
Ok c'est bon pour moi je merge mais je squash tes commits en un seul du coup
